DataStructures/Fenwick/BinaryIndexedTree.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+
+namespace DataStructures.Fenwick
+{
+    /// <summary>
+    /// Represent classical realization of Fenwik tree or Binary Indexed tree.
+    ///
+    /// BITree[0..n] --> Array that represents Binary Indexed Tree.
+    /// arr[0..n-1] --> Input array for which prefix sum is evaluated.
+    /// </summary>
+    public class BinaryIndexedTree
+    {
+        private readonly int[] bITree = new int[1000];
```

Comment:
Can't we make the size of that array a parameter and pass it through the constructor?

Reply:
Yes, we can. I will update the size of the tree based on the size of the input array.

Reply:
Replaced with List. Now tree size can non-fixed.